CSharp/Library/Microsoft.Bot.Connector.Shared/MicrosoftAppCredentials.cs

```diff
@@ -83,7 +83,17 @@ public MicrosoftAppCredentials(IConfiguration configuration, ILogger logger = nu
         public string MicrosoftAppId { get; set; }
         public string MicrosoftAppPassword { get; set; }
 
-        public virtual string OAuthEndpoint { get { return JwtConfig.ToChannelFromBotLoginUrl; } }
+        public virtual string OAuthEndpoint
+        {
+            get
+            {
+                string tenant = null;
+#if NET45
+                tenant = SettingsUtils.GetAppSettings("ChannelAuthTenant");

```

Comment:
Should be defensive about string.Empty. Also should indicate this section only applies for advanced users.